### ISSUE TITLE
Vint does not show warnings about using '!' and 'abort' for autoload functions

### ISSUE THREAD

**Issue by ranebrown** [dense-analysis/ale#2935]
## Information

**VIM version**
VIM - Vi IMproved 8.1 (2018 May 18, compiled Nov 26 2019 21:43:07)
macOS version
Included patches: 1-2350

Operating System: macOS Mojave 10.14.6

## What went wrong

When using the Vint linter ALE does not mark warning about functions in autload directories using `!` and `abort` e.g. `function! test#test() abort` instead of `function test#test()`.

## Reproducing the bug

Minimal vimrc:
```vim
set nocompatible
call plug#begin('~/.vim/plugged')
Plug 'dense-analysis/ale'
call plug#end()
syntax enable
filetype plugin indent on
```

Create a file `autoload/test/test.vim` with contents:
```vim
function test#test()
    let mystr = "Hello"
    return
endfunction
```

ALE marks line 2 with a warning about using single quoted strings.

Running vint with `vint -s .` produces:
```sh
autoload/test/test.vim:1:1: Use the abort attribute for functions in autoload (see Google VimScript Style Guide (Functions))
autoload/test/test.vim:2:17: Prefer single quoted strings (see Google VimScript Style Guide (Strings))
```

ALE should also be marking line 1 with the warning about using the `abort` attribute. I would guess this occurs because ALE is running on the temp file `var/folders/rx/0h60k4y14b5_4wxmbqv8j88w00zt_m/T/vfEmhdg/0/test.vim` so the `autoload` directory information is lost.

<details>
  <summary>:ALEInfo</summary>

 Current Filetype: vim
Available Linters: ['ale_custom_linting_rules', 'vint']
  Enabled Linters: ['ale_custom_linting_rules', 'vint']
 Suggested Fixers: 
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_vim_vint_executable = 'vint'
let g:ale_vim_vint_show_style_issues = 1
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) vint
(finished - exit code 0) ['/usr/local/bin/bash', '-c', '''vint'' --version']

<<<OUTPUT STARTS>>>
0.3.21
<<<OUTPUT ENDS>>>

(finished - exit code 1) ['/usr/local/bin/bash', '-c', '''vint'' -s --no-color -f "{file_path}:{line_number}:{column_number}: {severity}: {description} (see {reference})" ''/var/folders/rx/0h60k4y14b5_4wxmbqv8j88w00zt_m/T/vfEmhdg/0/test.vim''']

<<<OUTPUT STARTS>>>
/var/folders/rx/0h60k4y14b5_4wxmbqv8j88w00zt_m/T/vfEmhdg/0/test.vim:2:17: warning: Prefer single quoted strings (see Google VimScript Style Guide (Strings))
<<<OUTPUT ENDS>>>
</details>

**Comment by w0rp** [comment]
Nice catch. This issue can be fixed by using the newer `--stdin-display-name` option once it's available in a stable release of Vint. A Vint release hasn't been made in a while now, so someone might have to help the maintainers of Vint publish a new release with the option.
